src/core/etl/src/Flow/ETL/Function/HTMLQuerySelector.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Flow\ETL\Function;
+
+use Dom\{Element, HTMLDocument};
+use Flow\ETL\Row;
+
+final class HTMLQuerySelector extends ScalarFunctionChain
+{
+    public function __construct(
+        private readonly mixed $value,
+        private readonly ScalarFunction|string $selector,
+    ) {
+    }
+
+    public function eval(Row $row) : ?Element
+    {
+        if (\PHP_VERSION_ID < 80400) {
```

Comment:
the reasony why we always check if specific function/class exists instead of using hardcoded version constraints, is because when you check class/function polyfills can enable this. When you hardcode that it requires 8.4, even with polyfill this won't work. 

Please move towards checking if specific class/function/method exists every where

Reply:
Adjusted, got confused as there was a mention to enforce version check.